### PR TITLE
mariadb: 10.6.3 -> 10.6.5

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
https://mariadb.com/kb/en/mariadb-1064-release-notes/
https://mariadb.com/kb/en/mariadb-1065-release-notes/

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
Needed small fix - remove mysqltest and update mariadb-client packages. See https://github.com/MariaDB/server/commit/f55d53e26d510f0e85225714e932de7980a7cabb
```
diff --git a/pkgs/servers/sql/mariadb/default.nix b/pkgs/servers/sql/mariadb/default.nix
index 42137b57e8a..f9106cd7b85 100644
--- a/pkgs/servers/sql/mariadb/default.nix
+++ b/pkgs/servers/sql/mariadb/default.nix
@@ -92,7 +92,6 @@ common = rec { # attributes common to both builds
     rm "$out"/bin/{mariadb-config,mariadb_config,mysql_config}
     rm -r $out/include
     rm -r $out/lib/pkgconfig
-    rm -rf "$out"/share/aclocal
   '';

   passthru.mysqlVersion = "5.7";
@@ -132,8 +131,7 @@ client = stdenv.mkDerivation (common // {
   ];

   postInstall = common.postInstall + ''
-    rm -rf "$out"/share/doc
-    rm "$out"/bin/{mysqltest,mytop}
+    rm "$out"/bin/{mariadb-test,mysqltest}
     libmysqlclient_path=$(readlink -f $out/lib/libmysqlclient${libExt})
     rm "$out"/lib/{libmariadb${libExt},libmysqlclient${libExt},libmysqlclient_r${libExt}}
     mv "$libmysqlclient_path" "$out"/lib/libmysqlclient${libExt}
@@ -196,6 +194,7 @@ server = stdenv.mkDerivation (common // {
   '';

   postInstall = common.postInstall + ''
+    rm -rf "$out"/share/aclocal
     chmod +x "$out"/bin/wsrep_sst_common
     rm "$out"/bin/{mariadb-client-test,mariadb-test,mysql_client_test,mysqltest}
   '' + optionalString withStorageMroonga ''

```

**Review by Izorkin — APPROVED** [review]

**Comment by Izorkin** [comment]
Add PMEM Support:
```
diff --git a/pkgs/servers/sql/mariadb/default.nix b/pkgs/servers/sql/mariadb/default.nix
index 0034a831bfe..36e53f08c3e 100644
--- a/pkgs/servers/sql/mariadb/default.nix
+++ b/pkgs/servers/sql/mariadb/default.nix
@@ -3,7 +3,7 @@
 , libaio, libevent, jemalloc, cracklib, systemd, perl
 , bzip2, lz4, lzo, snappy, xz, zlib, zstd
 , fixDarwinDylibNames, cctools, CoreServices, less
-, numactl # NUMA Support
+, numactl, pmdk
 , withStorageMroonga ? true, kytea, libsodium, msgpack, zeromq
 , withStorageRocks ? true
 }:
@@ -150,7 +150,7 @@ server = stdenv.mkDerivation (common // {
     libxml2 judy libevent cracklib
   ] ++ optional (stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isAarch32) numactl
     ++ optionals withStorageMroonga [ kytea libsodium msgpack zeromq ]
-    ++ optional stdenv.hostPlatform.isLinux linux-pam
+    ++ optional stdenv.hostPlatform.isLinux [ linux-pam pmdk ]
     ++ optional (!stdenv.hostPlatform.isDarwin) mytopEnv;

   patches = common.patches;

```

**Comment by Izorkin** [comment]
@ajs124 check this patch:
```
diff --git a/pkgs/servers/sql/mariadb/default.nix b/pkgs/servers/sql/mariadb/default.nix
index 0034a831bfe..d3d543ae097 100644
--- a/pkgs/servers/sql/mariadb/default.nix
+++ b/pkgs/servers/sql/mariadb/default.nix
@@ -1,9 +1,15 @@
-{ lib, stdenv, fetchzip, fetchFromGitHub, cmake, pkg-config, makeWrapper, ncurses, nixosTests
-, libiconv, openssl, pcre2, boost, judy, bison, libxml2, libkrb5, linux-pam, curl
-, libaio, libevent, jemalloc, cracklib, systemd, perl
+{ lib, stdenv, fetchzip, fetchFromGitHub, nixosTests
+# Native buildInputs components
+, bison, boost, cmake, fixDarwinDylibNames, flex, makeWrapper, pkg-config
+# Common components
+, curl, libiconv, ncurses, openssl, pcre2
+, libaio, libkrb5, systemd
+, CoreServices, cctools, perl
+, jemalloc, less
+# Server components
 , bzip2, lz4, lzo, snappy, xz, zlib, zstd
-, fixDarwinDylibNames, cctools, CoreServices, less
-, numactl # NUMA Support
+, cracklib, judy, libevent, libxml2
+, linux-pam, numactl, pmdk
 , withStorageMroonga ? true, kytea, libsodium, msgpack, zeromq
 , withStorageRocks ? true
 }:
@@ -30,12 +36,13 @@ common = rec { # attributes common to both builds
   };

   nativeBuildInputs = [ cmake pkg-config ]
-    ++ optional stdenv.hostPlatform.isDarwin fixDarwinDylibNames;
+    ++ optional stdenv.hostPlatform.isDarwin fixDarwinDylibNames
+    ++ optional (!stdenv.hostPlatform.isDarwin) makeWrapper;

   buildInputs = [
-    ncurses openssl zlib pcre2 libiconv curl
-  ] ++ optionals stdenv.hostPlatform.isLinux [ libaio systemd libkrb5 ]
-    ++ optionals stdenv.hostPlatform.isDarwin [ perl cctools CoreServices ]
+    curl libiconv ncurses openssl pcre2 zlib
+  ] ++ optionals stdenv.hostPlatform.isLinux [ libaio libkrb5 systemd ]
+    ++ optionals stdenv.hostPlatform.isDarwin [ CoreServices cctools perl]
     ++ optional (!stdenv.hostPlatform.isDarwin) [ jemalloc ];

   prePatch = ''
@@ -93,6 +100,11 @@ common = rec { # attributes common to both builds
     rm -r $out/lib/pkgconfig
   '';

+  # perlPackages.DBDmysql is broken on darwin
+  postFixup = optionalString (!stdenv.hostPlatform.isDarwin) ''
+    wrapProgram $out/bin/mytop --set PATH ${makeBinPath [ less ncurses ]}
+  '';
+
   passthru.mysqlVersion = "5.7";

   passthru.tests = {
@@ -143,15 +155,15 @@ server = stdenv.mkDerivation (common // {

   outputs = [ "out" "man" ];

-  nativeBuildInputs = common.nativeBuildInputs ++ [ bison boost.dev ] ++ optional (!stdenv.hostPlatform.isDarwin) makeWrapper;
+  nativeBuildInputs = common.nativeBuildInputs ++ [ bison boost.dev flex ];

   buildInputs = common.buildInputs ++ [
     bzip2 lz4 lzo snappy xz zstd
-    libxml2 judy libevent cracklib
+    cracklib judy libevent libxml2
   ] ++ optional (stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isAarch32) numactl
-    ++ optionals withStorageMroonga [ kytea libsodium msgpack zeromq ]
-    ++ optional stdenv.hostPlatform.isLinux linux-pam
-    ++ optional (!stdenv.hostPlatform.isDarwin) mytopEnv;
+    ++ optionals stdenv.hostPlatform.isLinux [ linux-pam pmdk.dev ]
+    ++ optional (!stdenv.hostPlatform.isDarwin) mytopEnv
+    ++ optionals withStorageMroonga [ kytea libsodium msgpack zeromq ];

   patches = common.patches;

@@ -204,11 +216,6 @@ server = stdenv.mkDerivation (common // {
     rm -r "$out"/OFF
   '';

-  # perlPackages.DBDmysql is broken on darwin
-  postFixup = optionalString (!stdenv.hostPlatform.isDarwin) ''
-    wrapProgram $out/bin/mytop --set PATH ${makeBinPath [ less ncurses ]}
-  '';
-
   CXXFLAGS = optionalString stdenv.hostPlatform.isi686 "-fpermissive";
 });
 in mariadb
```
Small cleanup and moved `postFixup` to comon phase (needed to `${mysql-client}/bin/mytop`).
And added `flex`:
```
-- Performing Test MASK_LONGDOUBLE - Success
-- flex/lex not found!
-- Performing Test have_C__Wall__Wmissing_declarations
```

**Comment by ajs124** [comment]
@Izorkin can we maybe do that in a separate PR? Or can you open a PR against my PR?

**Comment by Izorkin** [comment]
Ok. I'll do another PR.

**Comment by Izorkin** [comment]
After merge this PR.

**Comment by vcunat** [comment]
This (probably) broke on x86_64-darwin and [will block](https://hydra.nixos.org/build/158130091#tabs-constituents) `nixpkgs-unstable`.  I just restarted it to try again to be sure, but the problem didn't seem transient.

**Comment by ajs124** [comment]
@NixOS/darwin-maintainers  ^

**Comment by vcunat** [comment]
I expect this is about which compiler you use, not so much about being darwin per se.  I'd probably bring these kind of issues upstream (maybe they have a ticket already).

**Comment by uri-canva** [comment]
The break was evident in the build in this PR too: https://github.com/NixOS/nixpkgs/runs/4179411814.
Please request help from `@NixOS/darwin-maintainers` before merging breakages, not after.

**Comment by ajs124** [comment]
@vcunat how about this:
```diff
diff --git a/pkgs/top-level/release.nix b/pkgs/top-level/release.nix
index fca5d376094..b4aeaadf2e8 100644
--- a/pkgs/top-level/release.nix
+++ b/pkgs/top-level/release.nix
@@ -45,7 +45,6 @@ let
               jobs.ghc.x86_64-darwin
               jobs.git.x86_64-darwin
               jobs.go.x86_64-darwin
-              jobs.mariadb.x86_64-darwin
               jobs.nix.x86_64-darwin
               jobs.nixpkgs-review.x86_64-darwin
               jobs.nix-info.x86_64-darwin
@@ -136,7 +135,6 @@ let
               jobs.nix-info.x86_64-darwin
               jobs.nix-info-tested.x86_64-darwin
               jobs.git.x86_64-darwin
-              jobs.mariadb.x86_64-darwin
               jobs.vim.x86_64-darwin
               jobs.inkscape.x86_64-darwin
               jobs.qt5.qtmultimedia.x86_64-darwin
```

it builds fine with clang 7 and 12 on x86_64-linux

**Comment by uri-canva** [comment]
Removing from blocking sounds good to me.

**Pull request merged**
